pkgs/development/python-modules/ffmpy/default.nix

```diff
@@ -0,0 +1,52 @@
+{ lib
+, buildPythonPackage
+, fetchFromGitHub
+, pythonOlder
+, pytestCheckHook
+, go
+, ffmpeg
+}:
+
+buildPythonPackage rec {
+  pname = "ffmpy";
+  version = "0.3.1";
+  format = "setuptools";
+
+  disabled = pythonOlder "3.6";
+
+  src = fetchFromGitHub {
+    owner = "Ch00k";
+    repo = "ffmpy";
+    rev = "refs/tags/${version}";
+    hash = "sha256-kuLhmCG80BmXdqpW67UanBnuYiL2Oh1jKt7IgmVNEAM=";
+  };
+
+  postPatch = ''
+    # default to store ffmpeg
+    substituteInPlace ffmpy.py \
+      --replace 'executable="ffmpeg",' 'executable="${ffmpeg}/bin/ffmpeg",'
+
+    # fix tests assuming "ffmpeg"
+    for fname in tests/*.py; do
+      echo 'FFmpeg.__init__.__defaults__ = ("ffmpeg", *FFmpeg.__init__.__defaults__[1:])' >>"$fname"
+    done
```

Comment:
We could just pass `ffmpeg` into `nativeCheckInputs`, no?

Reply:
I just tested this different scenarios

removing the `# fix tests ` code block (including the for loop) and adding ffmpeg to nativecheckinputs doesn't work unfortunately

The failures are of type
```
 AssertionError: assert '/nix/store/f...-exit-code 42' == 'ffmpeg --std...-exit-code 42'
ffmpy> E         - ffmpeg --stdin none --stdout none --stderr multiline --exit-code 42
ffmpy> E         + /nix/store/fy4nryi6sg44iy0ipq1j46vq8v9hq5v3-ffmpeg-5.1.3-bin/bin/ffmpeg --stdin none --stdout none --stderr multiline --exit-code 42
```

strangely enough, the tests pass when removing the following code
```
    # default to store ffmpeg
    substituteInPlace ffmpy.py \
      --replace 'executable="ffmpeg",' 'executable="${ffmpeg}/bin/ffmpeg",'
```

since I'm not sure how to properly test ffmpy, I suggest to leave as is.

Reply:
The tests test a mock that do not behave like ffmpeg. If we default to in-store ffmpeg they fail.

Reply:
I added that exact comment to clarify the situation.